tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -22,7 +22,9 @@
 #include "llvm/Support/raw_ostream.h"
 
 namespace {
-spv::ExecutionModel getSpirvShaderStageFromHlslProfile(const char *profile) {
+spv::ExecutionModel
+getSpirvShaderStageFromHlslProfile(clang::DiagnosticsEngine &diags,
```

Comment:
I'd suggest to keep `diags` as the second parameter since it is less important than `profile` and it reads more naturally of `...FromHlslProfile(...profile, ...)`.